admin/promotions/Promotions_Admin_Page.core.php

```diff
@@ -31,482 +32,586 @@ class Promotions_Admin_Page extends EE_Admin_Page
-            'promotion_exclusive' => EEH_Form_Fields::select_input('PRO_exclusive', $this->_yes_no_values, $this->_promotion->is_exclusive()),
-            'promotion_uses' => $promotion_uses !== EE_INF_IN_DB ? $promotion_uses : '',
+        $form_args             = [
+            'promotion'             => $this->_promotion,
+            'promotion_global'      => EEH_Form_Fields::select_input(
+                'PRO_global',
+                $this->_yes_no_values,
+                $this->_promotion->is_global()
+            ),
+            'promotion_exclusive'   => EEH_Form_Fields::select_input(
+                'PRO_exclusive',
+                $this->_yes_no_values,
+                $this->_promotion->is_exclusive()
+            ),
+            'promotion_uses'        => $promotion_uses !== EE_INF_IN_DB ? $promotion_uses : '',
             'promotion_global_uses' => $promotion_global_uses !== EE_INF_IN_DB ? $promotion_global_uses : '',
-            'price_type_selector' => $this->_get_price_type_selector(),
-            'scope_selector' => $this->_get_promotion_scope_selector()
-            );
-        EEH_Template::display_template(EE_PROMOTIONS_ADMIN_TEMPLATE_PATH . 'promotion_details_form.template.php', $form_args);
+            'price_type_selector'   => $this->_get_price_type_selector(),
+            'scope_selector'        => $this->_get_promotion_scope_selector(),
+            'promo_starts_datepicker' => new PromotionsDatepicker(
+                $this->_promotion->start('Y-m-d', 'h:i a'),
+                'start',
+                '#promotion-details-mbox',
+                'PRO_start',
+                esc_html__('enter promotion start date', 'event_espresso')
+            ),
+            'promo_ends_datepicker' => new PromotionsDatepicker(
+                $this->_promotion->end('Y-m-d', 'h:i a'),
+                'end',
+                '#promotion-details-mbox',
+                'PRO_end',
+                esc_html__('enter promotion end date', 'event_espresso')
+            ),
+        ];
+        EEH_Template::display_template(
+            EE_PROMOTIONS_ADMIN_TEMPLATE_PATH . 'promotion_details_form.template.php',
+            $form_args
+        );
     }
 
 
-
     /**
      * promotions_applied_to_metabox
      *
-     * @throws \EE_Error
+     * @throws EE_Error
+     * @throws ReflectionException
      */
     public function promotions_applied_to_metabox()
     {
         // we use the scope to get the metabox content.
         $scope = $this->_promotion->scope_obj();
 
         // if there is no scope then this is a default promotion object so the content will for promotions metabox will be generic.
-        $content =  empty($scope) ? __('When you select a scope for the promotion this area will have options related to the selection.', 'event_espresso') : $scope->get_admin_applies_to_selector($this->_promotion->ID());
+        $content = empty($scope) ? __(
```

Comment:
```suggestion
        $content = empty($scope) ? esc_html__(
```
A lot of these in this file. If even interested in fixing those here.